gtsam/sfm/MFAS.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ *  @file  MFAS.cpp
+ *  @brief Source file for the MFAS class
+ *  @author Akshay Krishnan
+ *  @date July 2020
+ */
+
+#include <gtsam/sfm/MFAS.h>
+
+using namespace gtsam;
+using std::pair;
+using std::vector;
+
+MFAS::MFAS(const std::shared_ptr<vector<Key>> &nodes,
+           const TranslationEdges& relativeTranslations,
+           const Unit3 &projection_direction)
+    : nodes_(nodes) {
+  // iterate over edges, obtain weights by projecting
+  // their relativeTranslations along the projection direction
+  for (auto it = relativeTranslations.begin();
+       it != relativeTranslations.end(); it++) {
+    edgeWeights_[it->first] = it->second.dot(projection_direction);
+  }
+}
+
+std::vector<Key> MFAS::computeOrdering() const {
```

Comment:
Maybe don’t go to four lines, but this function is indeed too long and complicated. Also ask yourself which STL algorithms you could be using.